### PR TITLE
[WIP] User refactor

### PR DESCRIPTION
**Pull request by rhutchison**
Up for discussion.

This is a work in progress. There is incomplete code and more to be done. I am looking for general direction feedback, not a code-review.

### CODE REVIEW

**Comment by mleanos** [comment]
@rhutchison You might consider breaking these changes up into multiple PR's. You have quite a bit going on here; with UI enhancements & back-end refactoring. It's getting a little unmanageable. It's going to hinder the ability for other developers to collaborate with you on this. I already experienced this for myself with this branch. 

You may gain some clarity, if you can *get back to nuetral* and give yourself a little more focus. Furthermore, these changes might get out quicker if they have a little more autonomy. I hope this helps; as I know you're deep in it right now.

File: scripts/reset-password.js [hunk 1]
```diff
@@ -1,50 +1,52 @@
 var nodemailer = require('nodemailer'),
-	mongoose = require('mongoose'),
-	config = require('../config/config'),
-	mg = require('../config/lib/mongoose');
+  mongoose = require('mongoose'),
```
Comment (ilanbiala) [inline scripts/reset-password.js]: These changes should be in a separate PR.

**Comment by lirantal** [comment]
@rhutchison another comment on the PR - can you please describe in a sentence/paragraph what this PR does? since this isn't some straight-forward, very specific and small PR, it requires someone to seriously code review everything to figure out all the bits and bites of what this PR does.

Would be helpful if you can explain it in simple words, what it affects, and what significant changes it makes so we can understand purpose and risks in this PR.

Thanks :)

**Comment by ilanbiala** [comment]
@rhutchison @mleanos @lirantal I think we're too far past this and this user implementation isn't much better than what @sielay had, so I'm just going to close and we can possibly work towards something better for 0.5.0 or 0.6.0.

**Comment by sielay** [comment]
@ilanbiala I have not time to sync with you, but copy what you want from CleanMean repo
